### PR TITLE
Genres api

### LINKED ISSUE #69: WebFlux: Data Access - Implement Genres DAO
## Description:
- Implement Functionality
   - GetAll & GetById
   - Reference dalGenres.cs in the C# project
- Implement Unit Tests

### PR DESCRIPTION
**Pull request by jyotsnaravikumar**
## Type of PR

- [ ] Documentation changes
- [ X] Code changes
- [ ] Test changes
- [ ] CI-CD changes
- [ ] GitHub Template changes

## Purpose of PR
/api/genres  API implementation

## Issues Closed or Referenced

- Closes #69

### CODE REVIEW

File: src/main/java/com/microsoft/cse/helium/app/controllers/GenresController.java [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+  /** getAllGenres. */
+  @RequestMapping(
+      value = "",
+      method = RequestMethod.GET,
+      produces = MediaType.APPLICATION_JSON_VALUE)
+  @ApiOperation(value = "Get all genres", notes = "Retrieve and return all genres")
+  @ApiResponses(value = {@ApiResponse(code = 200, message = "List of actor objects")})
+  public Mono<List<String>> getAllGenres() {
+    return genresDao.getGenres();
+  }
```
Comment (jofultz) [inline src/main/java/com/microsoft/cse/helium/app/controllers/GenresController.java]: I think this should be a Flux<> return type.

File: src/main/java/com/microsoft/cse/helium/app/dao/GenresDao.java [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+  }
+
+  /**
+   * getActors.
+   *
+   * @return
+   */
+  public Mono<List<String>> getGenres() {
+    logger.info("genreQuery " + genreQuery);
+
```
Comment (jofultz) [inline src/main/java/com/microsoft/cse/helium/app/dao/GenresDao.java]: Same here, I think it should be Flux<>.

Comment (jyotsnaravikumar) [reply src/main/java/com/microsoft/cse/helium/app/dao/GenresDao.java]: This is a workaround as CosmosDB SDK do not support value queries. To handle that scenario we need to convert Object Array to [{"genre":"action","genre":adventure"...}] to String Array from a stream ["action",adventure"...] .  CollectMap is the only way to handle stream of strings. I verified this with cosmos db team as well. We can move to return Flux only the "value" queries are supported by CosmosDB SDK

File: src/main/java/com/microsoft/cse/helium/app/controllers/GenresController.java (hunk 1 above)
Comment (jyotsnaravikumar) [reply src/main/java/com/microsoft/cse/helium/app/controllers/GenresController.java]: Same as mentioned in previous comment. Object Array is converted into String Array and collected as Single Item, hence it is a Mono<List>. Once Value Queries are supported we can use. Flux

**Comment by jyotsnaravikumar** [comment]
Closing PR , will be recreated

**Pull request closed without merging**